mypy/subtypes.py

```diff
@@ -1377,14 +1374,6 @@ def visit_overloaded(self, left: Overloaded) -> bool:
     def visit_union_type(self, left: UnionType) -> bool:
         return all([self._is_proper_subtype(item, self.orig_right) for item in left.items])
 
-    def visit_type_guard_type(self, left: TypeGuardType) -> bool:
-        if isinstance(self.right, TypeGuardType):
-            # TypeGuard[bool] is a subtype of TypeGuard[int]
-            return self._is_proper_subtype(left.type_guard, self.right.type_guard)
-        else:
-            # TypeGuards aren't a subtype of anything else for now (but see #10489)
-            return False
```

Comment:
So just to clarify, what is the view on subtyping between type guard functions? Can a type guard method be overridden in a subclass? Can a type guard function be expected as a callback?

Reply:
I believe yes to both.

But note that `mypy.types.TypeGuardedType` doesn't actually appear in the signature of `def f(x: object) -> TypeGuard[str]`. The return type is just treated as a bool https://github.com/python/mypy/blob/e1bd6cf3c06f5d96cd42fc86ed740586e278018d/mypy/typeanal.py#L356
`mypy.types.TypeGuardedType` is only created in find_isinstance_check: https://github.com/python/mypy/blob/9c9c3580fa80a7c4145b2951bd91c31dfa6bc7e6/mypy/checker.py#L4268